src/api/nlopt-in.hpp

```diff
@@ -89,10 +98,29 @@ namespace nlopt {
     typedef struct {
       opt *o;
       mfunc mf; func f; void *f_data;
+      functor_type functor;
       vfunc vf;
       nlopt_munge munge_destroy, munge_copy; // non-NULL for SWIG wrappers
     } myfunc_data;
 
+    static void* alloc_myfunc_data_with_nulls() {
+      // need to return void* otherwise SWIG doesn't compile because
+      // myfunc_data is private
```

Comment:
This is a private method, no?  So casts shouldn't be needed for SWIG.

Reply:
It is private right now, but I think you might be right, in the sense it wouldn't compile when it was public and then I both changed it to private and to `void*` returning.
Unfortunately, I have no way of testing this locally, could you run the CI please?